gulpfile.js

```diff
@@ -162,17 +174,142 @@ function cordovaCompile() {
   // cordova-ios@5.0.0. See https://github.com/apache/cordova-ios/issues/404.
   const compileArgs = platform === 'ios' ? '--device --buildFlag="-UseModernBuildSystem=0"' : '';
   const releaseArgs = isRelease ? platform === 'android' ?
-                                  `--release -- --keystore=${gutil.env.KEYSTORE} ` +
+                                  `--release --keystore=${gutil.env.KEYSTORE} ` +
               `--storePassword=${gutil.env.STOREPASS} --alias=${gutil.env.KEYALIAS} ` +
               `--password=${gutil.env.KEYPASS}` :
                                   '--release' :
                                   '';
-  return runCommand(`cordova compile ${platform} ${compileArgs} ${releaseArgs} -- ${platformArgs}`);
+  return runCommand(`yarn cordova compile ${platform} ${compileArgs} ${releaseArgs} ${platformArgs}`);
 }
 
 const setupWebApp = gulp.series(buildWebApp, transpileWebApp, writeEnvJson);
 const setupCordova = gulp.series(cordovaPlatformAdd, cordovaPrepare, xcode);
 
+//////////////////
+//////////////////
+//
+// Watch tasks
+//
+//////////////////
+//////////////////
+
+const cordovaBuild = gulp.series(cordovaPrepare, xcode, cordovaCompile);
+
+/**
+ * Trigger system notification when rebuild complete if `--notify` argument
+ * passed
+ */
+function notifyRebuildComplete(done) {
+  if (!watchNotify) {
+    return;
+  }
+
+  nodeNotifier.notify({
+    icon: path.join(__dirname, 'resources/icons/osx/icon-64.png'),
+    message: `${platform} rebuild complete`,
+    remove: 'ALL',
+    title: 'outline-client',
+  });
+
+  done();
+}
+
+/**
+ * If platform is browser and `--reload-macos-browser` argument passed, use
+ * AppleScript to reload front tab of browser. (Only works on macOS since it
+ * relies on AppleScript, and only works on Chromium and Safari browsers since
+ * Firefox has a weak AppleScript dictionary.)
+ *
+ * If platform is android and `--reinstall-android-app` arugment passed, use
+ * `adb` to reinstall and start Android app on attached device. (This may not
+ * work well on some configurations -- it has only been tested on one machine.)
+ */
+function reloadAppOrBrowser(done) {
+  if (
+    platform === 'browser'
+    && typeof watchReloadMacosBrowser === 'string'
+  ) {
+    const appleScriptCommand = (
+      watchReloadMacosBrowser.includes('Safari') && (
+        'set URL of front document to (URL of front document)'
+      )
+      || (watchReloadMacosBrowser.includes('Chrom') || watchReloadMacosBrowser.includes('Edge')) && (
+        'reload active tab of window 1'
+      )
+      || null
+    )
```

Comment:
This was difficult to read. How about a ternary?

```js
const appleScriptCommand =
  watchReloadMacosBrowser.includes('Safari')
  ? 'set URL of front document to (URL of front document)'
  : watchReloadMacosBrowser.includes('Chrom') || watchReloadMacosBrowser.includes('Edge')
  ? 'reload active tab of window 1
  : null;
```

Reply:
What do you think of [this](https://github.com/Jigsaw-Code/outline-client/pull/759/commits/b47ddc0e6d445f1356248b3ac65de799d9ea7c9a)?

I didn’t go with your ternary suggestion since Prettier moves the `?` and `:` to the end of lines, which IMO made it awkward to read. My way is longer, but it’s clearer than my original version, which I agree was unnecessarily confusing 😅.

P.S. I didn’t know/remember that ternary expressions could be chained like that — thanks for the tip!

Reply:
Well I'm fine with however Prettier arranges it! (I think syntax highlighting would still attract my attention to where it's needed.)

But yes I guess I prefer your new version!

(Sorry for slow response.)